python/plugins/processing/algs/qgis/voronoi.py

```diff
@@ -884,6 +906,29 @@ def computeDelaunayTriangulation(points):
         if len(args) > 0:
             fp.close()
 
+        if filtering and threshold > 0:
+
+            # [Home Page](http://github.com/karimbahgat/Pyqtree)
+            # filter input using spacial tree
+            from pyqtree import _QuadTree as Qtree
```

Comment:
should be manage if pyqtree is not installed... btw did you test using internal https://www.qgis.org/api/classQgsSpatialIndex.html class? there ore c++ processing algs that eavely use spatialindex and have massive speedup especially for static data like in this case @nyalldawson can give more details

Reply:
Yes - QgsSpatialIndex should work here and would be preferable to the added dependency.

Reply:
A little outside of this PR, but I think we should test the performance of our algorithms. In particular, this one - which must be widely used with Delaunay - should be compared with a C++ version (QGIS or GEOS).

Reply:
@lbartoletti i tend to agree - I'd love to use geos' method here. The issue is that geos method returns a purely spatial geometry, and loses the topological links back to the original features. We could potentially "rebuild" this after, but may run into precision issues when trying to link output vertices back to input ones. In any case, it's unfortunately not a trivial change.

Reply:
Was initially using this method outside of Qgis context, so have to find method to handle it with pure python solution. While available delaunay code was qgis one, i wanted to share this idea. https://youtu.be/UoneP16PaNc

At some point delaunay itself should be ported to c implementation as it remains the main bottleneck in this process and is pretty damn slow (scipy is arround 18x faster than pure python) - and c sources does exists. Using __slots__ help a bit both for speed and memory footprint.

But whatever the delaunay implementation, the input filtering remains usefull on badly sampled data sets, and this is the main purpose of this patch.

Pyqtree does perform quite well, the most obvious drawn back lie in init parameters requiring knowledge of center and boundary of data set - so if internal one doesn't require such pre-processing it really could worth the effort.

There is no need to store topological info into the tree itself, we only need to know if something already "hold a place" at desired location, and skip adding to site list when anything is found.